homeassistant/components/tado/climate.py

```diff
@@ -642,11 +749,23 @@ def _control_hvac(
             temperature_to_send = None
 
         fan_speed = None
+        fan_level = None
         if self.supported_features & ClimateEntityFeature.FAN_MODE:
-            fan_speed = self._current_tado_fan_speed
+            if self._current_tado_fan_speed in TADO_FAN_LEVELS:
+                fan_level = self._current_tado_fan_speed
+            elif self._current_tado_fan_speed in TADO_FAN_SPEEDS:
+                fan_speed = self._current_tado_fan_speed
         swing = None
+        vertical_swing = None
+        horizontal_swing = None
         if self.supported_features & ClimateEntityFeature.SWING_MODE:
-            swing = self._current_tado_swing_mode
+            if self._attr_swing_modes is not None:
```

Comment:
Can we reduce the nesting here? Can an early-reutn principle be used here?

Reply:
Yes you're right. I think I can merge it in a single if.
I don't think I can use a return because I don't want to skip the call to set_zone_overlay for that reason.

Reply:
Feel free to let me know when you committed these changes and I need to review them again.

Reply:
It's okay. Let me know if you want some other changes.